### ISSUE TITLE
Rabbitmq CrashLoopBackOff after injection with SmartDNS

### ISSUE THREAD

**Issue by FrimIdan** [istio/istio#30715]
**Bug description**
After applying istio on rabbitmq cluster the rabbitmq-server-0 enters a CrashLoopBackOff.
After disabling the smartDNS by setting `ISTIO_META_DNS_CAPTURE: "false"`  and redeploying the server again - **problem was not reproduce**

[crash.log](https://github.com/istio/istio/files/5939965/crash.log)
[no-crash.log](https://github.com/istio/istio/files/5939967/no-crash.log)

The error from the crashed container:
```
2021-02-02 15:35:54.064 [error] <0.272.0> Peer discovery returned an error: "{failed_connect,[{to_address,{\"kubernetes.default\",443}},\n                 {inet,[inet],nxdomain}]}". Will retry after a delay of 500 ms, 4 retries left...
.
.
.
BOOT FAILED
===========
Exception during startup:
    rabbit_boot_steps:run_boot_steps/1 line 20
2021-02-02 15:35:55.250 [info] <0.44.0> Application mnesia exited with reason: stopped
2021-02-02 15:35:55.250 [info] <0.44.0> Application mnesia exited with reason: stopped
2021-02-02 15:35:55.250 [error] <0.272.0> 
2021-02-02 15:35:55.250 [error] <0.272.0> BOOT FAILED
2021-02-02 15:35:55.250 [error] <0.272.0> ===========
2021-02-02 15:35:55.250 [error] <0.272.0> Exception during startup:
2021-02-02 15:35:55.251 [error] <0.272.0> 
2021-02-02 15:35:55.251 [error] <0.272.0>     rabbit_boot_steps:run_boot_steps/1 line 20
2021-02-02 15:35:55.251 [error] <0.272.0>     rabbit_boot_steps:'-run_boot_steps/1-lc$^0/1-0-'/1 line 19
    rabbit_boot_steps:'-run_boot_steps/1-lc$^0/1-0-'/1 line 19
    rabbit_boot_steps:run_step/2 line 46
    rabbit_boot_steps:'-run_step/2-lc$^0/1-0-'/2 line 41
    rabbit_mnesia:init/0 line 76
    rabbit_mnesia:init_with_lock/3 line 111
    rabbit_mnesia:run_peer_discovery_with_retries/2 line 145
    rabbit_mnesia:run_peer_discovery_with_retries/2 line 138
error:{badmatch,ok}
2021-02-02 15:35:55.253 [error] <0.272.0>     rabbit_boot_steps:run_step/2 line 46
2021-02-02 15:35:55.253 [error] <0.272.0>     rabbit_boot_steps:'-run_step/2-lc$^0/1-0-'/2 line 41
2021-02-02 15:35:55.253 [error] <0.272.0>     rabbit_mnesia:init/0 line 76
2021-02-02 15:35:55.253 [error] <0.272.0>     rabbit_mnesia:init_with_lock/3 line 111
2021-02-02 15:35:55.254 [error] <0.272.0>     rabbit_mnesia:run_peer_discovery_with_retries/2 line 145
2021-02-02 15:35:55.254 [error] <0.272.0>     rabbit_mnesia:run_peer_discovery_with_retries/2 line 138
2021-02-02 15:35:55.254 [error] <0.272.0> error:{badmatch,ok}
2021-02-02 15:35:55.254 [error] <0.272.0> 
2021-02-02 15:35:56.255 [info] <0.271.0> [{initial_call,{application_master,init,['Argument__1','Argument__2','Argument__3','Argument__4']}},{pid,<0.271.0>},{registered_name,[]},{error_info,{exit,{{badmatch,ok},{rabbit,start,[normal,[]]}},[{application_master,init,4,[{file,"application_master.erl"},{line,138}]},{proc_lib,init_p_do_apply,3,[{file,"proc_lib.erl"},{line,226}]}]}},{ancestors,[<0.270.0>]},{message_queue_len,1},{messages,[{'EXIT',<0.272.0>,normal}]},{links,[<0.270.0>,<0.44.0>]},{dictionary,[]},{trap_exit,true},{status,running},{heap_size,1598},{stack_size,28},{reductions,431}], []
2021-02-02 15:35:56.255 [error] <0.271.0> CRASH REPORT Process <0.271.0> with 0 neighbours exited with reason: {{badmatch,ok},{rabbit,start,[normal,[]]}} in application_master:init/4 line 138
2021-02-02 15:35:56.255 [info] <0.44.0> Application rabbit exited with reason: {{badmatch,ok},{rabbit,start,[normal,[]]}}
2021-02-02 15:35:56.255 [info] <0.44.0> Application rabbit exited with reason: {{badmatch,ok},{rabbit,start,[normal,[]]}}
Kernel pid terminated (application_controller) ({application_start_failure,rabbit,{{badmatch,ok},{rabbit,start,[normal,[]]}}})
{"Kernel pid terminated",application_controller,"{application_start_failure,rabbit,{{badmatch,ok},{rabbit,start,[normal,[]]}}}"}
Crash dump is being written to: /var/log/rabbitmq/erl_crash.dump...done
```

[ ] Docs
[ ] Installation
[X] Networking
[ ] Performance and Scalability
[ ] Extensions and Telemetry
[ ] Security
[ ] Test and Release
[ ] User Experience
[ ] Developer Infrastructure
[ ] Upgrade

**Expected behavior**
rabbitmq server should spin w/o restarts

**Steps to reproduce the bug**
istioctl install --set profile=preview
kubectl apply -f "https://github.com/rabbitmq/cluster-operator/releases/latest/download/cluster-operator.yml"
kubectl create ns rabbitmq && kubectl label ns rabbitmq istio-injection=enabled && kubectl -n rabbitmq  apply -f definition.yaml (attached)

[definition.txt](https://github.com/istio/istio/files/5939966/definition.txt)

**Version** (include the output of `istioctl version --remote` and `kubectl version --short` and `helm version --short` if you used Helm)
$ istioctl version --remote
client version: 1.8.2
control plane version: 1.8.2
data plane version: 1.8.2 (3 proxies)
$ kubectl version --short
Client Version: v1.19.4
Server Version: v1.16.15-gke.6000

**How was Istio installed?**
istioctl install --set profile=preview

**Environment where the bug was observed (cloud vendor, OS, etc)**
GKE

Additionally, please consider running `istioctl bug-report` and attach the generated cluster-state tarball to this issue.
Refer [cluster state archive](http://istio.io/help/bugs/#generating-a-cluster-state-archive) for more details.

**Comment by esnible** [comment]
There are two closed issues for RabbitMQ in #14743 .

**Comment by FrimIdan** [comment]
@esnible neither are related to my issue, or am I missing something?

**Comment by howardjohn** [comment]
I can reproduce, thanks for the details.

**Comment by howardjohn** [comment]
and cannot reproduce on 1.9 branch. unfortunately all the helpful logging that was added is on the 1.9 branch...

**Comment by FrimIdan** [comment]
So problem was solved in 1.9 but we don't know what? Was there changes regarding to SmartDNS that you think are related?

**Comment by howardjohn** [comment]
Actually does it *never* stop crashing? For me, I see it crash 2x then work. And the proxy takes 20s to get ready. So I am thinking its blocking dns requests for 20s -> crash. But once its ready, crash stops

**Comment by FrimIdan** [comment]
It does stop crashing, for me I saw sometimes after 5 restarts and some other time after ~20.
I also tried to set _holdApplicationUntilProxyStarts_ to _true_  but it does not help.

You need to re-install the server to reproduce it again, simple "delete pod" will not work.
You can delete the definition file.

**Comment by howardjohn** [comment]
> You need to re-install the server to reproduce it again, simple "delete pod" will not work.

ah, that was likely my problem

**Comment by howardjohn** [comment]
In my testing https://github.com/istio/istio/pull/29718  fixes it, which is in 1.8. When using preview profile WASM slows down init to 20s+, which makes the first few attempts crash. @FrimIdan will confirm 1.8.3 fixes

**Comment by FrimIdan** [comment]
@howardjohn 
I can confirm that problem was solved in 1.8.3.
Got a single restart and then app runs smootekly.
When configuring _holdApplicationUntilProxyStarts_ to _true_ - no restarts was observed
